### PR TITLE
[plugin.video.newyankeeworkshop] 0.0.3

### PR DESCRIPTION
**Pull request by th0ma5w**
### Description
  - Updated the plugin for Matrix to use Python 3.x
### Checklist:
- [X] My code follows the [add-on rules](http://kodi.wiki/view/Add-on_rules) and [piracy stance](http://kodi.wiki/view/Official:Forum_rules#Piracy_Policy) of this project. 
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/repo-plugins/blob/master/CONTRIBUTING.md) document
- [X] Each add-on submission should be a single commit with using the following style: [plugin.video.foo] v1.0.0

### CODE REVIEW

File: plugin.video.newyankeeworkshop/addon.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8" standalone="yes"?>
+<addon id="plugin.video.newyankeeworkshop" name="The New Yankee Workshop" version="0.0.3" provider-name="th0ma5w">
+	<requires>
+		<import addon="xbmc.python" version="3.0.0"/>
+		<import addon="xbmc.addon" version="18.9.701" />
```
Comment (enen92) [inline plugin.video.newyankeeworkshop/addon.xml]: please drop the xbmc.addon dependency

File: plugin.video.newyankeeworkshop/router.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    xbmcplugin.endOfDirectory(addon_handle)
+elif mode[0] == 'season':
+    season_link = args['season'][0]
+    episode_list = get_episode_list(season_link)
+    list_items=[]
+    for episode in episode_list:
+        episode_display, episode_link = episode
+        url = build_url({'mode': 'episode', 'season': season_link, 'episode': episode_link, 'display': episode_display})
+        li  = xbmcgui.ListItem(episode_display)
+        li.setArt({'icon': os.path.join(ADDON_PATH,'icon.png')})
```
Comment (enen92) [inline plugin.video.newyankeeworkshop/router.py]: maybe pass the whole dict to the setArt call? You'd avoid calling setArt multiple times which has a performance penalty.

**Review by enen92 — CHANGES_REQUESTED** [review]
See comments. Thanks for the update

**Comment by th0ma5w** [comment]
@enen92  All requested changes have been made. Thank you so much for the feedback, and I do think it is actually noticeably faster.

**Review by enen92 — APPROVED** [review]

**Pull request merged**